contracts/examples/DXswapSwapRelayer.sol

```diff
@@ -0,0 +1,296 @@
+pragma solidity =0.6.6;
+pragma experimental ABIEncoderV2;
+
+import './OracleCreator.sol';
+import './../interfaces/IDXswapFactory.sol';
+import './../interfaces/IDXswapRouter.sol';
+import './../libraries/TransferHelper.sol';
+import './../interfaces/IERC20.sol';
+import './../interfaces/IWETH.sol';
+import './../libraries/SafeMath.sol';
+import './../libraries/DXswapLibrary.sol';
+
+contract DXswapSwapRelayer {
+    using SafeMath for uint256;
+
+    event NewOrder(uint256 indexed _orderIndex);
+
+    event ExecutedOrder(uint256 indexed _orderIndex);
+
+    event WithdrawnExpiredOrder(uint256 indexed _orderIndex);
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    event Details(uint256 amountIn, uint256 amountOut);
+
+    struct Order {
+        address tokenIn;
+        address tokenOut;
+        uint256 amountIn;
+        uint256 priceTolerance;
+        uint256 minReserveA;
+        uint256 minReserveB;
+        address oraclePair;
+        uint256 deadline;
+        uint256 maxWindowTime;
+        uint256 oracleId;
+        address factory;
+        bool executed;
+    }
+
+    uint256 public immutable GAS_ORACLE_UPDATE = 168364;
```

Comment:
I'm not sure about this. The underlying protocol can change and call gas costs might be different in the future... using `gasleft()` to calculate the specific amount of gas used to perform the oracle update is a better way of doing things imho.